### ISSUE TITLE
Add withJvmArguments to Gradle Test Runner

### ISSUE THREAD

**Issue by JLLeitschuh** [gradle/gradle#5075]
In order to configure the JVM arguments used by the `GradleRunner` you must cast it to the `DefaultGradleRunner`. Shouldn't setting the JVM args get exposed through the the `GradleRunner`?

https://github.com/gradle/kotlin-dsl/blob/5c4702bfeafa1f78678c258019d184bce2a48d61/test-fixtures/src/main/kotlin/org/gradle/kotlin/dsl/fixtures/AbstractIntegrationTest.kt#L157

**Comment by ldaley** [comment]
The reason it was not done originally is that it was not obvious whether we would be able to abstract away from the build under test happening in a separate JVM. This intersects with the “debug” aspect of `GradleRunner` that explicitly inlines the build runtime into the current JVM therefore making it impossible to set JVM arguments for the build. 

The hope was that more real world usage would reveal what were the most important use cases to prioritize, and that would guide the design WRT making some of the internal features public. Others in the team may be able to answer whether that is now the case or not better than I can.

**Comment by bmuschko** [comment]
For reference we also had a longer discussion about this feature request here: https://github.com/gradle/gradle/issues/1043

**Comment by JLLeitschuh** [comment]
> We are not planning to expose this as public API as I am sure it will allow users to shoot themselves into their foot given the arguments described in my comment.

My concern comes from having people looking at the `kotlin-dsl` project (as I was) for hints as to how to setup their own test framework to test Kotlin DSL Gradle plugins and seeing the use of `DefaultGradleRunner`.

**Comment by JLLeitschuh** [comment]
Is this the "recommended" way of doing this for test developers?

**Comment by eskatos** [comment]
@JLLeitschuh it is the recommended way to set the JVM arguments of the Gradle daemon

Using a `org.gradle.jvmargs` property set in `gradle.properties` file has the same behavior as the internal `DefaultGradleRunner. withJvmArguments()`: JVM arguments are ignored when TestKit debug mode is enabled.

In other words, there's effectively a public mean to get the same behavior as this internal method.
The original request could be seen as "provide an API to do the same as with the public `org.gradle.jvmargs` property".

**Comment by JLLeitschuh** [comment]
Fair enough, makes sense. I'll consider this issue closed unless anyone else has anything else to discuss here.
